idea/idea-frontend-independent/src/org/jetbrains/kotlin/idea/quickfix/AddFunctionBodyFix.kt

```diff
@@ -30,9 +30,9 @@ class AddFunctionBodyFix(element: KtFunction) : KotlinPsiOnlyQuickFixAction<KtFu
         }
     }
 
-    companion object : KotlinSingleIntentionActionFactory() {
-        public override fun createAction(diagnostic: Diagnostic): AddFunctionBodyFix? {
-            return diagnostic.psiElement.getNonStrictParentOfType<KtFunction>()?.let(::AddFunctionBodyFix)
+    companion object : QuickFixesPsiBasedFactory<KtFunction>(KtFunction::class, PsiElementSuitabilityCheckers.ALWAYS_SUITABLE) {
+        override fun doCreateQuickFix(psiElement: KtFunction): List<IntentionAction> {
+            return listOfNotNull(psiElement.getNonStrictParentOfType<KtFunction>()?.let(::AddFunctionBodyFix))
```

Comment:
I know you didn't change this, but I'm a little confused here. If `psiElement` is the function without a body, why do we even need to get the parent? Shouldn't this just be `return listOf(AddFunctionBodyFix(psiElement)`?

I think `getNonStrictParentOfType` returns the argument if it matches the type (`KtFunction`) so the call doesn't even do anything.

Reply:
`getNonStrictParentOfType` https://github.com/JetBrains/intellij-community/blob/0f0b9c476078ee84d2579cc344f721ef38b839a5/platform/core-api/src/com/intellij/psi/util/PsiTreeUtil.java#L866

Reply:
Good call. I think it's not needed in this case since this change has narrowed the input from `PsiElement` to `KtFunction`. Looks like all existing usage are reporting on `KtFunction` as well so it should be safe to remove this call.